python/ml4ir/base/model/optimizer.py

```diff
@@ -7,17 +7,20 @@
 def get_optimizer(
     optimizer_key: str,
     learning_rate: float,
-    learning_rate_decay: float,
-    learning_rate_decay_steps: int,
-    gradient_clip_value: float,
+    learning_rate_decay: float = 1.0,
```

Comment:
Setting default values here.

Reply:
One of the drawback here is that we are overriding default tf values (with the same values, which makes things not robust/ideal).

Reply:
Yeah, I agree. Thoughts on how we should go about this? If I don't specify defaults, then it will expect us to pass this argument to the `get_optimizer` function each time. I can wrap it in if-else clauses, but it will be very dirty.

Reply:
The code here looks for different optimizers based on keys that are configured on `ml4ir.base.config.keys`. We can come up with a way that skips this, and given the users choice directly looks at tensorflow.keras.optimizers.USER_CHOICE, if exists it returns it, if not it returns SGD with a message that it was not found (or simply fails with the message). This removes the need to maintain `ml4ir.base.config.keys`, fizes the problem of default arguments as they will be maintained. Arguments can be passed as done here, all being optional apart from the optimizer_key. Further args, spacific to each optimizer, can be passed with the (ugly) **kwards. The last advantage is that we expose the full functionality of the optimizers to a user.

Reply:
@balikasg I'm not sure I know how to do this -> "looks at tensorflow.keras.optimizers.USER_CHOICE, if exists it returns it". I need an enum because when using ml4ir as a script people will enter a string. I need to a way to map that string to a subclass of `tensorflow.keras.optimizers`